Discovery endpoints must not request authentication

Currently, when I open https://openeo.mundialis.de/api/v0.3/ I am prompted:

![grafik](https://user-images.githubusercontent.com/17781682/55001971-052de400-4fd6-11e9-803f-4aeb556546fc.png)

One should be able to discover a backend without the need to sign up. I.e. the endpoints

- /
- /collections
- /collections/{collection_id}
- /processes
- /output_formats
- /service_types
- /udf_runtimes (new in v0.4)

should be reachable without authentication.

For example, this is essential for the Hub -- password-protected backends can't be crawled.

We also get CORS errors when things are requested against these `undefined` locations, like this:

```
Access to XMLHttpRequest at 'https://openeo.mundialis.de/api/v0.4/jobs/undefined' from origin 'https://open-eo.github.io' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Credentials' header in the response is '' which must be 'true' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
utils.js:9 Error: Network Error
    at e.exports (createError.js:16)
    at XMLHttpRequest.h.onerror (xhr.js:81)
exception @ utils.js:9
(anonymous) @ JobPanel.vue:153
Promise.catch (async)
createJob @ JobPanel.vue:152
(anonymous) @ JobPanel.vue:164
save @ ParameterModal.vue:58
ne @ vue.runtime.esm.js:1854
n @ vue.runtime.esm.js:2179
Qi.o._wrapper @ vue.runtime.esm.js:6911
xhr.js:172 GET https://openeo.mundialis.de/api/v0.4/jobs/undefined net::ERR_FAILED
```

--- a/src/openeo_grass_gis_driver/udf.py
+++ b/src/openeo_grass_gis_driver/udf.py
@@ -100,6 +100,6 @@ GET_UDF_DOC = {
 }
 
 
-class Udf(ResourceBase):
+class Udf(Resource):
     def get(self):
         return make_response(jsonify(SUPPORTED_UDF), 200)

